workbench-core/example/infrastructure/integration-tests/tests/datasets/list.test.ts

```diff
@@ -24,15 +25,143 @@ describe('datasets list integration test', () => {
   });
 
   describe('ListDataSets', () => {
-    it('should return DataSets entries', async () => {
-      const response = await adminSession.resources.datasets.get();
-      expect(response.data).toBeDefined();
+    let datasets: Dataset[];
+    let userId: string;
+    let groupId: string;
+    let updatedAdminSession: ClientSession;
+
+    beforeAll(async () => {
+      // create a dynamic authz group and add the authenticated user to it
+      const { data } = await adminSession.resources.groups.create();
+      groupId = data.groupId;
+      userId = adminSession.getSettings().get('rootUserId');
+      await adminSession.resources.groups.group(groupId).addUser({ userId });
+
+      // create adminSession with updated authenticated user groups
+      updatedAdminSession = await setup.createAdminSession();
+    });
+
+    beforeEach(async () => {
+      // create a group of 5 datasets
+      const dataSetsData = await Promise.all(
+        Array.from({ length: 5 }, () => adminSession.resources.datasets.create())
+      );
+      datasets = dataSetsData.map(
+        (data) => adminSession.resources.datasets.children.get(data.data.id) as Dataset
+      );
+
+      // remove default permission from each dataset
+      await Promise.all(
+        datasets.map((dataset) =>
+          dataset.removeAccess({
+            permission: {
+              identity: userId,
+              identityType: 'USER',
+              accessLevel: 'read-only'
+            }
+          })
+        )
+      );
+    });
+
+    it('should return an empty array when the user has no permissions on any dataset', async () => {
```

Comment:
We need at least one test that covers each of the unexpected cases where the user has CREATE, UPDATE, DELETE access (IdentityPermissions will need to be created directly through authz), but not READ access. We also need a case here where a user has the 'ALLOW' effect on one READ permission and a 'DENY' on another.